smack-experimental/src/main/java/org/jivesoftware/smackx/mam/provider/MamPrefsIQProvider.java

```diff
@@ -63,6 +71,9 @@ public MamPrefsIQ parse(XmlPullParser parser, int initialDepth, XmlEnvironment x
             case START_ELEMENT:
                 final String name = parser.getName();
                 switch (name) {
+                case "prefs":
+                    elementFactory = MamVersion.fromNamespace(parser.getNamespace()).newElementFactory();
```

Comment:
As above, I am not sure, just from reading the code, why this is needed. Can you elaborate what's going on here? We don't do this dance in `MamQueryIQProvider` for example.

Reply:
As the parser initially returns `null` in the `MamPrefIQProviderTest` we call `getNamespace()` again when we encounter the `<prefs>` element.